app/src/main/kotlin/io/scalac/android/circlegame/engine/Engine.kt

```diff
@@ -1,14 +1,155 @@
 package io.scalac.android.circlegame.engine
 
+import android.os.Handler
+import android.os.HandlerThread
+import android.os.Message
+import android.os.SystemClock
 import io.scalac.android.circlegame.model.CircleModel
 import timber.log.Timber
+import kotlin.properties.Delegates
 
-interface EngineView {
-    fun showCircle(circle: CircleModel)
+interface LevelChangedListener {
+    fun onLevelChanged(currentLevel: Int)
 }
 
-class Engine(val engineView: EngineView) {
+interface EngineView : LevelChangedListener {
```

Comment:
Tak, bo chciałem żeby Level miał możliwość poinformowania przez ten interfejs o zmianie levelu. Nie chciałem tam przekazywać EngineView to level wiedziałby za dużo.